var/spack/repos/builtin/packages/cuda-toolkit/package.py

```diff
@@ -165,6 +165,24 @@ class Cuda(Package):
     # see https://docs.nvidia.com/cuda/archive/11.4.0/cuda-gdb/index.html#release-notes
     depends_on('ncurses abi=5', type='run', when='@:11.3.99+dev')
 
+    provides('cuda@11.6')
+    provides('cuda@11.5', when='@:11.5.1')
```

Comment:
Why the range? You want a specific version to provide a specific version.

Reply:
The logic I had in mind was something along this line:
```py
v = '11.6' # default version
if c > '11.4.4':
    v = '11.5'
elif c > `11.3.1':
    v = '11.4'
elif ...
```
to account for the case when more than one patchlevel (`y` in, say, `12.x.y`) are published per minor version `x`,
If you take a quick look at [CUDA Toolkit archive](https://developer.nvidia.com/cuda-toolkit-archive) currently there are `11.6.0`, `11.6.1`, and `11.6.2` available for download.

Reply:
You should just make `cuda-toolkit@x.y.z` provide `cuda@x.y.z`, there are conflicts on the patch version number with gcc, those should still be respected.